### PR TITLE
ARROW-2205: [Python] Option for integer object nulls

### PR DESCRIPTION
**Pull request by adshieh**
Fixes [ARROW-2205](https://issues.apache.org/jira/browse/ARROW-2205).

### CODE REVIEW

**Comment by wesm** [comment]
Thanks for working on this @adshieh! I am wondering out loud if there's anything we can do to help with the API for a growing number of pandas conversion arguments (like using an options object instead of keyword args)

**Comment by xhochy** [comment]
@wesm I think using `kwargs` seems to be the most pythonic way to do this. With Pandas I also wondered in the beginning over the large number of kwargs but in the end, it seems like a good-enough solution.

**Comment by pitrou** [comment]
>  I am wondering out loud if there's anything we can do to help with the API for a growing number of pandas conversion arguments (like using an options object instead of keyword args)

Perhaps the dialect concept used by the [csv module](https://docs.python.org/3/library/csv.html) can be re-used?

**Comment by adshieh** [comment]
I personally prefer keyword arguments because the number of calls to `to_pandas` in my use cases has been limited, so having the documentation in the method and avoiding the extra step of creating an options object has been convenient.

File: python/pyarrow/tests/test_convert_pandas.py [hunk 1]
```diff
@@ -615,6 +615,36 @@ def test_int_object_nulls(self):
         _check_pandas_roundtrip(df, expected=expected,
                                 expected_schema=schema)
 
+    def test_int_object_nulls_option(self):
```
Comment (cpcloud) [inline python/pyarrow/tests/test_convert_pandas.py]: It doesn't look like you're using `self` here. Can you make this into a test function and [`pytest.mark.parametrize`](https://docs.pytest.org/en/latest/parametrize.html) it on the `int_dtypes` parameter?

Comment (wesm) [reply python/pyarrow/tests/test_convert_pandas.py]: We should probably convert this whole module to pytest-style in a separate patch

Comment (adshieh) [reply python/pyarrow/tests/test_convert_pandas.py]: Sure! However, it seems like none of the test methods use `self` and the test classes are just for organizational purposes, so moving it to a test function would be a deviation?

Comment (cpcloud) [reply python/pyarrow/tests/test_convert_pandas.py]: Sure, but if we are going to do it eventually then we shouldn't knowingly add to the debt in the name of consistency.

Comment (pitrou) [reply python/pyarrow/tests/test_convert_pandas.py]: I would vote against the pytest-style of a forest of functions. In my experience the lack of organization produces difficult to maintain test modules. Organizing the test methods into several classes helped me figure out which features were tested and how.

An alternative would be to split the tests into several modules (or perhaps several modules inside a subpackage).

Comment (cpcloud) [reply python/pyarrow/tests/test_convert_pandas.py]: Grouping by modules (which contain functions) is the solution to that particular problem.

**Review by xhochy — APPROVED** [review]
+1, LGTM

would like to see another approval before merging as we had some discussions on the options

**Comment by wesm** [comment]
yes, plan to review this today

**Review by wesm — APPROVED** [review]
+1, this LGTM. Thank you @adshieh!

**Comment by wesm** [comment]
Rebased. Going to wait for the builds to rujn

**Comment by wesm** [comment]
Rebasing this again

**Pull request closed without merging**